src/main/java/bio/terra/service/snapshotbuilder/SnapshotBuilderService.java

```diff
@@ -131,9 +134,21 @@ public SnapshotBuilderGetConceptsResponse searchConcepts(
       UUID datasetId, String domainId, String searchText, AuthenticatedUserRequest userRequest) {
     Dataset dataset = datasetService.retrieve(datasetId);
     TableNameGenerator tableNameGenerator = getTableNameGenerator(dataset, userRequest);
+    SnapshotBuilderSettings snapshotBuilderSettings =
+        snapshotBuilderSettingsDao.getSnapshotBuilderSettingsByDatasetId(datasetId);
+
+    SnapshotBuilderDomainOption snapshotBuilderDomainOption =
+        snapshotBuilderSettings.getDomainOptions().stream()
+            .filter(domainOption -> Objects.equals(domainOption.getCategory(), domainId))
```

Comment:
There is no need to use `Objects.equals()` here unless you're worried that `domainOption.getCategory()` will be `null`. Since we're getting it from `snapshotBuilderSettings` we should be able to assert that it won't be `null` and we can call `equals()` on it directly.
```suggestion
            .filter(domainOption -> domainOption.getCategory().equals(domainId))
```